backend/mysql_utils.js

```diff
@@ -0,0 +1,228 @@
+    let sql = "SELECT records.id, records.creator, records.name,  DATE_FORMAT(records.publish_date, '%Y-%m-%d %r') publish_date, records.link, users.email_id FROM records LEFT JOIN users ON records.creator=users.id";
+    if (!isAdmin) sql += ` WHERE creator='${userId}'`;
+
+    mysqlConnPool.getConnection(function (err, conn) {
+        if (!err) {
+            conn.query(sql, function (err, results) {
+                conn.release();
+                if (!err) {
+                    let records = {};
+                    if (results.length) {
+                        results.forEach(function (item, index) {
+                            records[item.id] = {
+                                id: item.id,
+                                name: item.name,
+                                publishDate: item.publish_date,
+                                link: item.link,
+                                creator: (item.email_id) ? item.email_id : item.creator,
+                            }
+                        });
+                        return callback(null, records);
+                    } else {
+                        return callback(null, null);
+                    }
+                } else {
+                    return dbError(err, callback);
+                }
+            });
+        } else {
+            return dbError(err, callback);
+        }
+    });
+}
+
+exports.getAllUsers = function (mysqlConnPool, callback) {
+    const sql = "SELECT id, email_id, is_active, is_admin FROM users";
+    mysqlConnPool.getConnection(function (err, conn) {
+        if (!err) {
+            conn.query(sql, function (err, results) {
+                conn.release();
+                if (!err) {
+                    let users = {};
+                    if (results.length) {
+                        results.forEach(function (item, index) {
+                            users[item.id] = {
+                                id: item.id,
+                                emailId: item.email_id,
+                                isActive: item.is_active,
+                                isAdmin: item.is_admin
+                            }
+                        });
+                    }
+                    return callback(null, users);
+                } else {
+                    console.error(`error getting users from database: ${err.toString()}`)
+                    return callback(err, null);
+                }
+            });
+        } else {
+            return dbError(err, callback);
+        }
+    });
+
+}
+
+
+exports.deleteUser = function (userId, mysqlConnPool, callback) {
+    const sql = `DELETE FROM users WHERE id=${userId}`;
+    mysqlConnPool.getConnection(function (err, conn) {
+        if (!err) {
+            conn.query(sql, function (err, result) {
+                conn.release();
+                if (err) {
+                    console.error(`error deleting user: ${err}`)
```

Comment:
Since we are doing things like this everywhere else
```suggestion
                    console.error(`error deleting user:${err.toString()}`)
```

Reply:
done

Reply:
done